src/NCronJob/Scheduler/TaskScheduler/TaskFactoryProvider.cs

```diff
@@ -0,0 +1,13 @@
+namespace NCronJob;
+
+internal static class TaskFactoryProvider
```

Comment:
I am much in favor of not commiting those 4 objects (all 3 schedulers and the factory).
If we need them we have the git history.

Reply:
I made some changes, kept only the Deterministic scheduler and added a config to ConcurrencySettings to be able to control which one to use.
Also for the `TaskFactoryProvider`, we still need to create a `TaskFactory` with those specific options. This provides cleaner code and SOC.